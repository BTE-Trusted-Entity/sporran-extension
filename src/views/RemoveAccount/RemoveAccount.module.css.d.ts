declare namespace RemoveAccountModuleCssNamespace {
  export interface IRemoveAccountModuleCss {
    address: string;
    addressLabel: string;
    buttonsLine: string;
    cancel: string;
    container: string;
    explanation: string;
    heading: string;
    name: string;
    remove: string;
    subline: string;
  }
}

declare const RemoveAccountModuleCssModule: RemoveAccountModuleCssNamespace.IRemoveAccountModuleCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: RemoveAccountModuleCssNamespace.IRemoveAccountModuleCss;
};

export = RemoveAccountModuleCssModule;
