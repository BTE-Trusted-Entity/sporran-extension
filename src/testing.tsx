export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
