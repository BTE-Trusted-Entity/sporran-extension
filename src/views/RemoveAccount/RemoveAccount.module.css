.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
  margin: 0;
}

.subline {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 0 0 0.75rem 0;
  line-height: 1.25rem;
}

.name {
  composes: accountText from '../../components/Typography/Typography.module.css';
  margin: 0.5rem 0 0.25rem 0;
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.addressLabel {
  composes: smallText from '../../components/Typography/Typography.module.css';
  margin: 0.5rem 0;
}

.address {
  composes: inputText from '../../components/Typography/Typography.module.css';
  margin: 0;
}

.explanation {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 1.5rem 0;
}

.buttonsLine {
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: center;
}

.cancel {
  composes: buttonPrimary from '../../components/Button/Button.module.css';
  margin: 0 1rem;
}

.remove {
  composes: buttonSecondary from '../../components/Button/Button.module.css';
  padding: 0 1rem;
}
