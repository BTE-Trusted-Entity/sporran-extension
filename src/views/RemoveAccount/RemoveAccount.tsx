import { useCallback } from 'react';
import { browser } from 'webextension-polyfill-ts';
import { Link } from 'react-router-dom';

import { Account } from '../../utilities/accounts/types';
import { Balance } from '../../components/Balance/Balance';
import { removeAccount } from '../../utilities/accounts/accounts';
import { LinkBack } from '../../components/LinkBack/LinkBack';
import { Stats } from '../../components/Stats/Stats';
import { paths } from '../paths';

import styles from './RemoveAccount.module.css';

interface Props {
  account: Account;
}

export function RemoveAccount({ account }: Props): JSX.Element {
  const t = browser.i18n.getMessage;

  const handleClick = useCallback(async () => {
    await removeAccount(account);
  }, [account]);

  return (
    <main className={styles.container}>
      <h1 className={styles.heading}>{t('view_RemoveAccount_heading')}</h1>
      <p className={styles.subline}>{t('view_RemoveAccount_subline')}</p>

      <p>{account.name}</p>
      <Balance address={account.address} />

      <p className={styles.addressLabel}>{t('view_RemoveAccount_address')}</p>
      <p className={styles.address}>{account.address}</p>
      <p className={styles.explanation}>
        {t('view_RemoveAccount_explanation')}
      </p>

      <Link to={paths.home} className={styles.cancel}>
        {t('common_action_cancel')}
      </Link>
      <button type="button" onClick={handleClick} className={styles.remove}>
        {t('view_RemoveAccount_remove')}
      </button>

      <LinkBack />

      <Stats />
    </main>
  );
}
