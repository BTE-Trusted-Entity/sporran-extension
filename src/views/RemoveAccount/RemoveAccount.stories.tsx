import { Meta } from '@storybook/react';

import { accountsMock as accounts } from '../../utilities/accounts/AccountsProvider.mock';

import { RemoveAccount } from './RemoveAccount';

export default {
  title: 'Views/RemoveAccount',
  component: RemoveAccount,
} as Meta;

const account = accounts['4tJbxxKqYRv3gDvY66BKyKzZheHEH8a27VBiMfeGX2iQrire'];

export function Template(): JSX.Element {
  return <RemoveAccount account={account} />;
}
