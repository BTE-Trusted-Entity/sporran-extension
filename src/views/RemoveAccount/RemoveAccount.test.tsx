import { render } from '../../testing';
import { RemoveAccount } from './RemoveAccount';

const account = {
  name: 'My Sporran Account',
  tartan: 'MacFarlane',
  address: '4tJbxxKqYRv3gDvY66BKyKzZheHEH8a27VBiMfeGX2iQrire',
  index: 1,
};

describe('RemoveAccount', () => {
  it('should render', () => {
    const { container } = render(<RemoveAccount account={account} />);
    expect(container).toMatchSnapshot();
  });
});
