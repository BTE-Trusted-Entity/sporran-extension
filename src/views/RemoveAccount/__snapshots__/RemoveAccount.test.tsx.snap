// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveAccount should render 1`] = `
<div>
  <main
    class="container"
  >
    <h1
      class="heading"
    >
      Attention!
    </h1>
    <p
      class="subline"
    >
      Youâ€™re about to remove your account
    </p>
    <p>
      My Sporran Account
    </p>
    <p
      class="balanceLine"
    >
      Total Balance: 
      <span
        class="amount"
      >
        1.2340
         
        <span
          aria-label="Kilt coin"
          class="coin"
        />
      </span>
    </p>
    <p
      class="addressLabel"
    >
      My account address
    </p>
    <p
      class="address"
    >
      4tJbxxKqYRv3gDvY66BKyKzZheHEH8a27VBiMfeGX2iQrire
    </p>
    <p
      class="explanation"
    >
      The next step will remove your account from the browser extension. This will not delete your account. You can add it again any time by importing your backup phrase.
    </p>
    <a
      class="cancel"
      href="/"
    >
      Cancel
    </a>
    <button
      class="remove"
      type="button"
    >
      Forget this account
    </button>
    <button
      aria-label="Back"
      class="linkBack"
      title="Back"
      type="button"
    />
    <p
      class="stats"
    >
      You have 3 accounts with a total
      <span
        class="balance"
      >
         Balance: 
        <span
          class="amount"
        >
          3.7020
           
          <span
            aria-label="Kilt coin"
            class="coin"
          />
        </span>
      </span>
    </p>
  </main>
</div>
`;
